### ISSUE TITLE
Rule enhancement: vue/max-attributes-per-line

### ISSUE THREAD

**Issue by sense-it-gmbh** [vuejs/eslint-plugin-vue#955]
**Please describe what the rule should do:**

In special cases it can be handy to be able to ignore the [vue/max-attributes-per-line](https://github.com/vuejs/eslint-plugin-vue/blob/master/docs/rules/max-attributes-per-line.md) rule for certain attributes.

For example when using [Vuetify's Grid](https://v15.vuetifyjs.com/en/framework/grid) there are the attributes for sizing, following the pattern `/(?:xs|sm|md|lg|xl)\d{1,2}/`. Putting each one on one line leads to bloated code (see example 1). To avoid bloated code it is handy to put them all on a single line, other attributes still on one line each (example 1, 2).

Putting all on a single line leads to an error. Allowing 5 Attributes per line makes the rule pointless.

With an extra option for an array of attributes and/or patterns that should be ignored by the rule would be handy.

**What category should the rule belong to?**

- [x] Enforces code style
- [ ] Warns about a potential error
- [ ] Suggests an alternate way of doing something
- [ ] Other (please specify:)

**Enforced syntax, leading to bloated code with `"multiline": 1`:**
```vue
<v-layout wrap>
  <v-flex
    v-for="n in 4" 
    fill-height
    xs12 
    md6 
    lg3
  >
    {{ n }}
  </v-flex>
</v-layout>
```

**Designated syntax, leading to an error with `"multiline": 1`:**
```vue
<v-layout wrap>
  <v-flex
    v-for="n in 4"
    fill-height
    xs12 md6 lg3
  >
    {{ n }}
  </v-flex>
</v-layout>
```

**Designated syntax, causing no error with `multiline: 1, ignore: '/(?:xs|sm|md|lg|xl)\d{1,2}/'`:**
```vue
<v-layout wrap>
  <v-flex
    v-for="n in 4" 
    fill-height
    xs12 md6 lg3
  >
    {{ n }}
  </v-flex>
</v-layout>
```

**Additional context**

The ignore pattern could be used in the root of the rule options or for each option separate:

```json
{
  "vue/max-attributes-per-line": ["error", {
    "singleline": 1,
    "multiline": {
      "max": 1,
      "allowFirstLine": false
    },
    "ignore": ["some-attribute", "/(?:xs|sm|md|lg|xl)\\d{1,2}/"]
  }]
}
```
```json
{
  "vue/max-attributes-per-line": ["error", {
    "singleline": {
      "max": 1,
      "ignore": ["some-attribute", "/(?:xs|sm|md|lg|xl)\\d{1,2}/"]
    },
    "multiline": {
      "max": 1,
      "allowFirstLine": false,
      "ignore": ["some-attribute", "/(?:xs|sm|md|lg|xl)\\d{1,2}/"]
    }
  }]
}
```

**Comment by ota-meshi** [comment]
Thank you for posting this.

I think perhaps your proposed option is hard to get the intended behavior when combined with `vue/attributes-order` rule.
I think you will be strongly dependent on the framework you use to get the behavior you intended.

Also, Vue team recommends one attribute per line.
https://vuejs.org/v2/style-guide/#Multi-attribute-elements-strongly-recommended

So I don't want to add this option to `vue/max-attributes-per-line` rule in eslint-plugin-vue.

**Comment by sense-it-gmbh** [comment]
Thats sad but i can understand your point.

**Comment by ota-meshi** [comment]
I close this issue because adding option is difficult with eslint-plugin-vue.
You can also create a new ESLint plugin and make your own rules and autofixes.

Thank you for understanding.
